fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -430,7 +430,8 @@ public CommandProcessingResult updateClient(final Long clientId, final JsonComma
             }
 
             final ExternalId externalId = externalIdFactory.createFromCommand(command, ClientApiConstants.externalIdParamName);
-            if (command.isChangeInStringParameterNamed(ClientApiConstants.externalIdParamName, externalId.getValue())) {
+            if (command.isChangeInStringParameterNamed(ClientApiConstants.externalIdParamName,
+                    clientForUpdate.getExternalId().getValue())) {
```

Comment:
Why do we need to use the `clientForUpdate` instead of the `externalId` variable?

Reply:
This is an issue in the client update, In the comparation there is not difference and then If you change the external Id value It is not updated

Reply:
Would you please raise a separate PR for this? In my understanding it has not much to do with the fix for updating datatable entry properly.